sql/sql_connect.cc

```diff
@@ -687,6 +687,41 @@ static bool login_connection(THD *thd) {
   DBUG_RETURN(0);
 }
 
+static void thd_update_net_stats(NET *net) noexcept {
+  if (net->last_errno == 0) {
+    return;
+  }
+
+  mysql_mutex_assert_not_owner(&LOCK_status);
+  mysql_mutex_lock(&LOCK_status);
```

Comment:
Can we make all of the statistics variables atomic, similar to connection_errors_select? Then we don't need to grab the LOCK_status() during connection cleanup for this.